src/common/snippets/include/snippets/lowered/pass/optimize_domain.hpp

```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "snippets/lowered/pass/pass.hpp"
+#include "snippets/shape_types.hpp"
+
+namespace ov {
+namespace snippets {
+namespace lowered {
+namespace pass {
+
+/**
+ * @interface OptimizeDomain
+ * @brief Collapse input/output dimensions to balance parallel/per-thread load
```

Comment:
Could you please add more detailed explanation here. Couple of thinks I would expect: 
1. Small graph example which shows how shapes are changed
2. Brief internal logic explanation (like how input params are used)

Reply:
Yes, certainly. Both added.